### PR TITLE
hand ranker

### PR DESCRIPTION
**Pull request by MaxwellChalmers**
# What is included in this change

implemented the hand ranking 

# Describe at least one problem you solved

typo's mostly, I kinda just went ahead and wrote a working solution for the ranks of hands with little to  no regard for finding the best way to do things, I thought the way I dealt with aces for straights was clever but I think it just made my code for that section look cluttered without really making things much easier to implement 

# How did you test this change?

running on docker, console.log statements and sending code to chatGPT to find syntax errors

# If things are not currently working, explain.

Need to do more testing to make sure there aren't issues but haven't seen bugs at this point 

# UI Changes 
Screen Shot 2024-02-18 at 2 40 54 PM

### CODE REVIEW

File: src/server.py [hunk 1]
```diff
@@ -104,4 +104,155 @@ async def discard_cards(deck_id: str, cards: str):
     
     return {"message": "Cards discarded successfully"}
 
+
+
+@app.get("/api/v2/highestRank/~{cards}")
```
Comment (semmypurewal) [inline src/server.py:109]: What does the `~` do in the route here?

File: src/server.py [hunk 2]
```diff
@@ -104,4 +104,155 @@ async def discard_cards(deck_id: str, cards: str):
...
+        return {"message": "Two Pair"}
+    if isPair(hand):
+        return {"message": "One Pair"}
+    
+    return {"message": "High Card"}
+
+def isStraightFlush(hand: list[Card]) -> bool:
+    isstraight = isStraight(hand)
+    isflush = isFlush(hand)
+    return isstraight and isflush
```
Comment (semmypurewal) [inline src/server.py:134]: Nice!

File: src/server.py [hunk 3]
```diff
@@ -104,4 +104,155 @@ async def discard_cards(deck_id: str, cards: str):
...
+    suit = hand[0].getSuit() 
+    for c in hand:
+        if c.getSuit() != suit:
+            flush = False
+            break
+    
+    return flush
+
+# a true abomination of coding style but I think it will work           
+def isStraight(hand: list[Card]) -> bool:
```
Comment (semmypurewal) [inline src/server.py:147]: This would be a good opportunity to write a test, but we haven't gotten to that part of the class yet :)

**Review by semmypurewal — APPROVED** [review]
This looks great! We'll talk more about ways of doing this as the semester goes on, so you might find some easier ways to achieve all of these things.

File: src/server.py (hunk 1 above)
Comment (MaxwellChalmers) [reply src/server.py:109]: I was getting a certain error and that seemed to fix the problem, might have been a different change that fixed the problem but I don't remember tbh.

**Pull request merged**